.github/workflows/build.yml

```
@@ -30,5 +31,5 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
       - run: python -m pip install --upgrade pip setuptools wheel
-      - run: python -m pip install --upgrade --editable .
-      - run: python setup.py test
+      - run: python -m pip install --upgrade --editable . -r requirements-dev.txt
```

The `[test]` extra can also be used here and in any instructions for running tests.

```suggestion
      - run: python -m pip install --upgrade --editable .[test] -r requirements-dev.txt
```

---

All the other requirements from `requirements-dev.txt` could be defined as `[dev]` extra and the command could be:

```
python -m pip install --upgrade --editable .[test,dev]
```

I'd also advise against using `--editable` on the CI. It is mostly also semi-deprecated / things are broken with PEP517/PEP518.

---

Do you mean to simply use
```
python -m pip install --upgrade .[test,dev]
```
?

---

yes

---

Thanks, I learned something today :)